### PR TITLE
Added a stringify function to avoid unneeded recursion when scanning code.

### PR DESCRIPTION
**Pull request by zinigor**
When an invocation is complex enough, it will go further than our logic in the analyzer, but we don't need to follow the properties or class names based on other variables - we still won't find out whether the class or property was moved in Jetpack without runtime analysis. Therefore we replace these instances with just a node's class name for want of a better string.

Fixes analyzer fatal errors on scanning various codebases.

#### Changes proposed in this Pull Request:

* Added a stopgap stringifier function that will let us avoid unnecessary recursion.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is part of the Jetpack DNA effort.

#### Testing instructions:

This change was prompted by scanning the entire WordPress repository using [the slurper plugin](https://github.com/markjaquith/WordPress-Plugin-Directory-Slurper), but you can see the fatals by scanning any code that uses the following code invocations: `new $this->class_name()`, `new $variable[ $class_name_index ]`, etc.
* Try scanning code that fails with missing property errors.
* Try scanning it with this fix, see that there's no failures anymore.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

File: packages/analyzer/src/PersistentList.php [hunk 1]
```diff
@@ -32,7 +32,13 @@ public function output() {
 	/**
 	 * Saves the items to a file and returns the file contents
 	 */
-	public function save( $file_path ) {
+	public function save( $file_path, $allow_empty = true ) {
```
Comment (gravityrail) [inline packages/analyzer/src/PersistentList.php]: You have an unrelated change here.

**Review by gravityrail — APPROVED** [review]
I'm going to approve this, but do note that you have an unrelated change in this PR. It's a nice improvement though, so I'm ok with it ;)

**Review by dereksmart — APPROVED** [review]

**Comment by dereksmart** [comment]
cherry-picked to feature branch 9b0e52b7b

**Comment by jeherve** [comment]
Cherry-picked to `branch-7.5` in 6494b55ac2ce5d14e64fd15c7215ce82c95c1f45
